### PR TITLE
chore: extract shared code between regionizers

### LINKED ISSUE #130: Extract common code shared between regionizers
This part is shared in H3 and S2 regionizer. Extracting it to `BaseRegionizer` is the right way IMHO

https://github.com/srai-lab/srai/blob/2bd619ae6d22f7569012847363f83850f87128f3/srai/regionizers/s2_regionizer.py#L67-L70

### PR DESCRIPTION
**Pull request by piotrgramacki**
closes #130

### CODE REVIEW

File: srai/regionizers/base.py [hunk 1]
```diff
@@ -23,3 +23,29 @@ def transform(self, gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
...
+        Set the CRS to WGS84.
+
+        Args:
+            gdf (gpd.GeoDataFrame): GeoDataFrame to be converted.
+
+        Returns:
+            GeoDataFrame converted to EPSG:4326.
+
+        """
+        return gdf.to_crs(epsg=4326)
```
Comment (Calychas) [inline srai/regionizers/base.py:38]: I would get rid of this function as it is a bit confusing (set vs to). I would opt for specifying library-wide constant crs="epsg:4326"

**Review by Calychas — CHANGES_REQUESTED** [review]

File: srai/regionizers/base.py (hunk 1 above)
Comment (piotrgramacki) [reply srai/regionizers/base.py]: I added this constant to `srai/utils/constants.py`. I used it everywhere except tests and example notebooks - if you disagree with that let me know.

File: srai/utils/constants.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+"""Constants used across the project."""
+
+CRS = "EPSG:4326"
```
Comment (RaczeQ) [inline srai/utils/constants.py:3]: Maybe we can change it to `WGS84_CRS` to make it more verbose.

File: srai/regionizers/voronoi_regionizer.py [hunk 3]
```diff
@@ -12,8 +12,11 @@
 
 from srai.utils._optional import import_optional_dependencies
 
+from ..utils.constants import CRS
```
Comment (RaczeQ) [inline srai/regionizers/voronoi_regionizer.py:15]: I know it's a valid import but I'm more keen on using a full import in this situation.
```suggestion
from srai.utils.constants import CRS
```

File: srai/utils/constants.py (hunk 2 above)
Comment (piotrgramacki) [reply srai/utils/constants.py:3]: I thought about having a default CRS for our library, so I didn't want to include its name in the constant. Do you think that such a default is a good idea or is it some overengineering?

Comment (RaczeQ) [reply srai/utils/constants.py:3]: I think it's not a very big change, but for everyone looking at the code from the outside, it'll be obvious which CRS we are using, without outside knowledge written in documentation or the readme file.

Comment (piotrgramacki) [reply srai/utils/constants.py:3]: Fair point, you convinced me

**Review by Calychas — APPROVED** [review]

**Pull request merged**